cluster/watchdog.go

```diff
@@ -121,6 +121,9 @@ func (w *Watchdog) rescheduleContainers(e *Engine) {
 			}
 		}
 
+		// Clear out the network configs, since we cannot create a
+		// container with multiple networks.
+		c.Config.NetworkingConfig.EndpointsConfig = nil
```

Comment:
When the network configs is wiped, `newContainer` wouldn't get this config. The re-created `newContainer` would have a different `NetworkingConfig.EndpointsConfig`. I suspect `newContainer` should restore the original's `Config.NetworkingConfig.EndpointsConfig`. Can you take a look?